### ISSUE TITLE
Ideas?

### ISSUE THREAD

**Issue by dy** [validate-io/ndarray#1]
Hi!

I would like to use this package in [glsl-transpiler](https://github.com/stackgl/glsl-transpiler) to detect n-dim array for texture.

What is supposed way to do so? Are you targeting [ndarray](https://npmjs.org/package/ndarray) specifically or you detect n-dim array in some generic case? If so, just interested how? I used to check `width`/`height`/`shape` properties.

Thanks!

**Comment by kgryte** [comment]
@dfcreative Yeah, essentially, I recommend duck-typing, as done [here](https://github.com/validate-io/matrix-like/blob/master/lib/index.js) for matrices. `ndarray` is just an interface, which any `object` data structure should be able to implement. As such, I would eschew using instance checks, etc.
